taffy/main.go

```diff
@@ -54,6 +55,20 @@ Options:
 		"taffy_job": os.Getenv("FLYNN_JOB_ID"),
 	}
 
+	if os.Getenv("HOME") == "" || os.Getenv("HOME") == "/" {
+		os.Setenv("HOME", "/root/")
+	}
+
+	if os.Getenv("SSH_CLIENT_KEY") != "" {
+		os.Mkdir(os.Getenv("HOME")+"/.ssh/", 0700)
+		ioutil.WriteFile(os.Getenv("HOME")+"/.ssh/id_rsa", []byte(os.Getenv("SSH_CLIENT_KEY")), 0600)
+	}
+
+	if os.Getenv("SSH_CLIENT_HOSTS") != "" {
+		os.Mkdir(os.Getenv("HOME")+"/.ssh/", 0700)
+		ioutil.WriteFile(os.Getenv("HOME")+"/.ssh/known_hosts", []byte(os.Getenv("SSH_CLIENT_HOSTS")), 0600)
```

Comment:
The code would probably be more readable if you put the env vars in variables, and you should use `filepath.Join` and `os.MkdirAll`:

```go
home := os.Getenv("HOME")

if key := os.Getenv("SSH_CLIENT_KEY"); key != "" {
  if err := os.MkdirAll(filepath.Join(home, ".ssh"); err != nil {
    log.Fatal(err)
  }
  if err := ioutil.WriteFile(filepath.Join(home, ".ssh", "id_rsa"), []byte(key), 0600); err != nil {
    log.Fatal(err)
  }
}